azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/AbstractAzureManager.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.lib.appservice.service.impl;
+
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.HasId;
+import com.microsoft.azure.arm.resources.ResourceId;
+import org.apache.http.HttpStatus;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public abstract class AbstractAzureManager<T extends HasId> {
+
+    protected boolean isRefreshed = false;
+
+    @Nonnull
+    protected String name;
+    @Nonnull
+    protected String resourceGroup;
+    @Nonnull
+    protected String subscriptionId;
+
+    protected T remote;
+
+    public AbstractAzureManager(@Nonnull final String id) {
+        final ResourceId resourceId = ResourceId.fromString(id);
+        this.name = resourceId.name();
+        this.resourceGroup = resourceId.resourceGroupName();
+        this.subscriptionId = resourceId.subscriptionId();
+    }
+
+    public AbstractAzureManager(@Nonnull final String subscriptionId, @Nonnull final String resourceGroup, @Nonnull final String name) {
+        this.name = name;
+        this.resourceGroup = resourceGroup;
+        this.subscriptionId = subscriptionId;
+    }
```

Comment:
can be replaced with `@RequiredArgsConstructor`(need be careful about the order of constructor args.)